### ISSUE TITLE
[DApp] Pagination broken on search page

### ISSUE THREAD

**Issue by franckc** [OriginProtocol/origin#742]
Tested on the mainnet:  Clicking on search icon, without any search terms returns a page with results but does not show bottom pagination to go to next page, even though the back-end server returned totalNumberOfItems=61

URL to reproduce: https://dapp.originprotocol.com/#/search?search_query=&listing_type=all

**Comment by franckc** [comment]
CC @micahalcorn @sparrowDom

**Comment by franckc** [comment]
Here is request to the server:
```
"
    {
      listings (
        searchQuery: ""
        filters: []
        page:{
          offset: 0
          numberOfItems: 12
        }
      ) {
        nodes {
          id
        }
        offset
        numberOfItems
        totalNumberOfItems
        stats {
          maxPrice
          minPrice
        }
      }
    }"

```
Here is the response from the server:
`{"data":{"listings":{"nodes":[{"id":"1-000-55"},{"id":"1-000-25"},{"id":"1-000-35"},{"id":"1-000-30"},{"id":"1-000-57"},{"id":"1-000-56"},{"id":"1-000-64"},{"id":"1-000-74"},{"id":"1-000-6"},{"id":"1-000-2"},{"id":"1-000-13"},{"id":"1-000-18"}],"offset":0,"numberOfItems":12,"totalNumberOfItems":61,"stats":{"maxPrice":1000,"minPrice":0.00005}}}}`

**Comment by micahalcorn** [comment]
@franckc only 12 results, and 3 of them are hidden. So the 9 that are showing is what I would expect. 😕

**Comment by micahalcorn** [comment]
So I don't think this is a pagination issue. I think it's a consequence of either listings not getting indexed or somehow not being send in the `search` result object.

**Comment by franckc** [comment]
The back-end only returns 12 listings because that is what is specified in the query via the "numberOfItems" field. But note that the response includes the field totalNumberOfItems set to 61 which indicates there are more listings to fetch. I verified by going to the graphQL UI at https://discovery.originprotocol.com/ and I then issued queries with offset 0, 12, 24, etc... and back-end returned all pages as expected.

@sparrowDom  wonder if the DApp code does not interpret totalNumberOfItems properly ?

**Comment by franckc** [comment]
I suspect the issue is here:
https://github.com/OriginProtocol/origin/blob/master/origin-dapp/src/components/listings-grid.js#L76

If renderMode is search, shouldn't this be using listingsLength ?
`const resultsCount = search.listingsLength`

**Comment by micahalcorn** [comment]
@franckc yep, blame me, sorry. I think it should be more like `resultsCount = search.listingsLength - hiddenListingIds.length` for the search mode and still `resultsCount = listingIdsNotHidden.length` for the other. But that screws up the pagination again.

**Comment by sparrowDom** [comment]
Sorry for seeing this issue a bit late. I see @micahalcorn has figured out the problem and prepared a solution for it.
